.github/workflows/codeql.yml

```diff
@@ -40,7 +40,7 @@ jobs:
     - uses: actions/setup-java@v4
       with:
         distribution: 'zulu'
-        java-version: '8'
+        java-version: '11'
```

Comment:
why 11 by default instead of 17? I was hoping we'd have 21 officially supported by now but I'm not sure it is yet which would be my real preference to have as the default since its the latest lts

Reply:
> why 11 by default instead of 17?

I thought we wanted to move to the immediate next version, instead of skipping versions. Also, was discussing this with Karan, and we weren't sure of how Druid built with 17 might behave on Java 11 environment.

Reply:
i think we should strive to default to the most recent lts version we support, we should be deprecating java 11 support soon too tbh, its getting quite old too

if we're worried about compiled with java 17 and running on 11 we could add make one of the mixed version things configured in this manner?